### DISCUSSION TITLE
smooth_shape and fair seem to have discontinuities with constrained regions

### DISCUSSION THREAD

**Discussion by pentacular** [CGAL/cgal#6756] [Q&A]
I'm doing a corefinement with an area, then a refine within the area, then a smooth_shape with all vertices not in that volume constrained.

This produces a plausible result, but I can't figure out why there's a sharp discontinuity between the constraint vertices and the unconstrained vertices.

I expected that it would try to minimize the curvature there as well, but I see a sharp edge where it collapses inward before producing the smooth corner that I expected.

Does anyone know of a good way to fair or smooth a subregion of a mesh while respecting the shape at the boundary of that region?

![Example](https://user-images.githubusercontent.com/33278307/178907084-ced7dbb5-1a8e-406c-8535-15126cab139e.png)

**Comment by janetournois** [comment]
Hello,
have you tried to use [`fair()`](https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__meshing__grp.html#ga9988e9a2eaf92740b3009d06c792f2cd) instead?
